src/LoginDialog/OidcUtilities.ts

```diff
@@ -0,0 +1,31 @@
+const generateStateToken = () => {
+    return Math.random().toString(36).substr(2) + Math.random().toString(36).substr(2);
```

Comment:
https://stackoverflow.com/questions/5651789/is-math-random-cryptographically-secure

We should not be using Math.random() for this, and we sholud make sure that we have 160 bits of entropy:

https://tools.ietf.org/html/rfc6749#section-10.10

Reply:
I am going to replace the random generator with this:
https://developer.mozilla.org/en-US/docs/Web/API/Crypto/getRandomValues

This will avoid bringing in another dependancy.